drizzlepac/haputils/hapcut_utils.py

```diff
@@ -0,0 +1,383 @@
+    if not isinstance(cutout_size, Quantity):
+        cutout_size *= u.arcsec
+        cutout_size = np.atleast_1d(cutout_size)
+        if len(cutout_size) == 1:
+            cutout_size = np.repeat(cutout_size, 2)
+
+    if not isinstance(sky_coord, SkyCoord):
+        sky_coord = SkyCoord(sky_coord, unit="deg")
+
+    # From HST data, Search for the list of images based upon: coordinates, search region, data
+    # product type, and the instrument name (with wildcard), project (HAP), and observation
+    # collection (HST).  Use the wildcard to get all the detectors for the instrument.  Multiple
+    # instruments cannot be searched at the same time.  Use the diagonal of the cutout to define
+    # the search radius for the archive.  Images which fall outside the desired cutout need to
+    # be filtered from the solution later.
+    radius = math.ceil(math.sqrt(math.pow(cutout_size.value[0], 2) + math.pow(cutout_size.value[1], 2)) / 2.0)
+
+    # Careful - the radius must be a str or Quantity
+    radius *= u.arcsec
+    if verbose:
+        print("Performing query for ACS images. Radius: {}.".format(radius))
+
+    # Note: calib_level does not seem to work
+    acs_query_table = Observations.query_criteria(coordinates=sky_coord,
+                                                  radius=radius,
+                                                  dataproduct_type="IMAGE",
+                                                  instrument_name="ACS*",
+                                                  project="HAP", 
+                                                  calib_level=3,
+                                                  obs_collection="HST") 
+
+    if verbose:
+        print("Performing query for WFC3 images.")
+    wfc3_query_table = Observations.query_criteria(coordinates=sky_coord,
+                                                   radius=radius,
+                                                   dataproduct_type="IMAGE",
+                                                   instrument_name="WFC3*",
+                                                   project="HAP", 
+                                                   calib_level=3,
+                                                   obs_collection="HST") 
+
+    query_table = vstack([acs_query_table, wfc3_query_table])
+
+    # Compute the limits of the cutout region
+    deg_cutout_size = cutout_size.to(u.deg)
+    ra_min = sky_coord.ra.value - deg_cutout_size.value[0]
+    ra_max = sky_coord.ra.value + deg_cutout_size.value[0]
+    dec_min = sky_coord.dec.value - deg_cutout_size.value[1]
+    dec_max = sky_coord.dec.value + deg_cutout_size.value[1]
+
+    # Filter the output as necessary to include only MVM filenames (MVM prefix: hst_skycell).
+    # Also, filter out images which are not actually in the requested cutout region as the
+    # archive search had to be done using a radius.
+    good_rows = []
+    updated_table = None
+    for i, old_row in enumerate(query_table):
```

Comment:
Does 'i' get used at all (I couldn't see where it did)?  If not, you could remove the use of 'enumerate' just to keep the code a little simpler.

Reply:
Residual cruft has been removed.